lnrpc/marshall_utils.go

```diff
@@ -0,0 +1,52 @@
+package lnrpc
+
+import (
+	"errors"
+
+	"github.com/btcsuite/btcutil"
+	"github.com/lightningnetwork/lnd/lnwire"
+)
+
+var (
+	// ErrSatMsatMutualExclusive is returned when both a sat and an msat
+	// amount are set.
+	ErrSatMsatMutualExclusive = errors.New(
+		"sat and msat arguments are mutually exclusive",
+	)
+)
+
+// CalculateFeeLimit returns the fee limit in millisatoshis. If a percentage
+// based fee limit has been requested, we'll factor in the ratio provided with
+// the amount of the payment.
+func CalculateFeeLimit(feeLimit *FeeLimit,
+	amount lnwire.MilliSatoshi) lnwire.MilliSatoshi {
+
+	switch feeLimit.GetLimit().(type) {
+	case *FeeLimit_FixedSat:
+		return lnwire.NewMSatFromSatoshis(
+			btcutil.Amount(feeLimit.GetFixedSat()),
+		)
```

Comment:
style nit: don't we usually add a newline after the body of a case block?

Reply:
fixed